processor/src/system/mod.rs

```diff
@@ -56,6 +56,12 @@ impl System {
         self.clk_trace.len()
     }
 
+    /// Returns free memory pointer from a clk cycle
```

Comment:
nit: I'd change the comment to "Returns free memory pointer at the specified clock cycle."